numpy/_core/overrides.py

```diff
@@ -19,12 +19,13 @@
         compatible with that passed in via this argument."""
 )
 
-def set_array_function_like_doc(public_api):
-    if public_api.__doc__ is not None:
-        public_api.__doc__ = public_api.__doc__.replace(
-            "${ARRAY_FUNCTION_LIKE}",
-            array_function_like_doc,
-        )
+def get_array_function_like_doc(public_api, docstring_template=None):
```

Comment:
Would be nice to mention the registration in the name.  But if we don't have a quick idea, no worries.

Reply:
agreed, I don't love that this function does two things, happy to take suggestions for a better name or pattern, but I don't have one now.